uat/component_init.feature

```diff
@@ -0,0 +1,58 @@
+Feature: gdk component init works
+
+  @version(min='1.0.0')
+  Scenario: init template with non empty dir
+    Given we have cli installed
+    When we run init command with -t,HelloWorld,-l,python
+    Then command was unsuccessful
+    And command output should have Try `gdk component init --help`
+
+  @version(min='1.0.0')
+  @change_cwd
+  Scenario: init template with empty dir
+    Given we have cli installed
+    When we run init command with -t,HelloWorld,-l,python
+    Then command was successful
+    And we verify component files
+
+  @version(min='1.1.0')
+  @change_cwd
+  Scenario: init template with new dir
+    Given we have cli installed
+    When we run init command with -t,HelloWorld,-l,python,-n,test-dir
+    Then command was successful
+    And we change directory test-dir
+    And we verify component files
+
+  @version(min='1.0.0')
+  @change_cwd
+  Scenario: init repository
+    Given we have cli installed
+    When we run init command with -r,aws-greengrass-labs-database-influxdb
+    Then command was successful
+    And we verify component files
+
+  @version(min='1.1.0')
+  @change_cwd
+  Scenario: init repository with new dir
+    Given we have cli installed
+    When we run init command with -r,aws-greengrass-labs-database-influxdb,-n,test-dir
+    Then command was successful
+    And we change directory test-dir
```

Comment:
nit:  Can we call directory as gdk project directory? then both `we change directory` and `we change gdk project directory` can point to the same step implementation.

Reply:
There is no step called `we change gdk project directory`, this would always change current directory.

I think you are confusing this with verification steps:
- `we verify gdk project files`
- `we verify component build files`
- `we verify component zip build files`

Reply:
Right, there's no step already but I was suggesting that we add one, so anyone who sees the UATs can understand that we are changing to gdk project directory to run commands rather than assuming that we are changing to some directory in the middle of a UAT. It's just a nit. I'm fine leaving it either way.

Reply:
I would keep as is